pyproject.toml

```diff
@@ -89,7 +89,7 @@ optional = [
   "scikit-learn>=0.24.2",
   # optuna/visualization/param_importances.py.
   "scipy",  # optuna/samplers/_gp
-  "torch",  # optuna/samplers/_gp
+  "torch; python_version<='3.12'", # TODO(gen740): Remove this line after torch supports Python 3.13
```

Comment:
The information that `torch` is required by `optuna/samplers/_gp` is missing in this PR.
Including a comment such as '# scipy and torch are required for optuna/samplers/_gp' would ensure that other contributors know the necessary modules' functionality packages.

(I'm not sure if writing dependencies in comments is a good practice. For example, I don't track whether those dependencies are updated when used in other modules.)

Reply:
In fact, also `optuna/_gp` has `torch` as a dependency (and also terminator, because it depends on `optuna/_gp`). So I think it is not realistic to list them all.

I realize that the above comment about 'scipy' is incorrect. It is not 'optuna/samplers/_gp' that uses 'scipy,' but rather 'optuna/_gp.'

I would comment like this `# TODO(gen740): Remove this line when 'torch', a dependency of 'optuna/_gp', supports Python 3.13` What do you think about this @toshihikoyanase?

Reply:
> I would comment like this # TODO(gen740): Remove this line when 'torch', a dependency of 'optuna/_gp', supports Python 3.13

Sounds reasonable. It will minimize the change while keeping the dependency information. Thanks!
(If you have a line-length issue, please keep the comment you proposed originally.)

> In fact, also optuna/_gp has torch as a dependency (and also terminator, because it depends on optuna/_gp). So I think it is not realistic to list them all.
> I realize that the above comment about 'scipy' is incorrect. It is not 'optuna/samplers/_gp' that uses 'scipy,' but rather 'optuna/_gp.'

Hmm, we may want to discuss the policy of comments to manage dependencies. But it is definitely out of scope of this PR.